botorch/acquisition/max_value_entropy_search.py

```diff
@@ -310,15 +403,186 @@ def _compute_information_gain(
+        batch of design points (X_pending)
+
+        Args:
+            X: A `batch_shape x 1 x d`-dim Tensor of `batch_shape` t-batches
+                with `1` `d`-dim design point each.
+            mean_M, variance_M: `batch_shape x 1`-dim Tensors of
+                `batch_shape` t-batches
+                All are obtained without noise.
+            covar_mM: `batch_shape x 1 x (1 + num_trace_observations)`
+                -dim Tensor.
+                All are obtained without noise.
+
+        Returns:
+            A `batch_shape`-dim Tensor of MVE values at the given design points `X`.
+        """
+
+        # TODO: give the Posterior API an add_observation_noise function to avoid
+        # doing posterior computations twice
+
+        # compute the mean_m, variance_m with noisy observation
+        posterior_m = self.model.posterior(X, observation_noise=True)
+        mean_m = self.weight * posterior_m.mean.squeeze(-1)
+        # batch_shape x 1
+        variance_m = posterior_m.variance.clamp_min(CLAMP_LB).squeeze(-1)
+        # batch_shape x 1
+        check_no_nans(variance_m)
+
+        # get stdv of noiseless variance
+        stdv = variance_M.sqrt()
+        # batch_shape x 1
+
+        # define normal distribution to compute cdf and pdf
+        normal = torch.distributions.Normal(
+            torch.zeros(1, device=X.device, dtype=X.dtype),
+            torch.ones(1, device=X.device, dtype=X.dtype),
+        )
+
+        # prepare max value quantities required by GIBBON
+        mvs = torch.transpose(self.posterior_max_values, 0, 1)
+        # 1 x s_M
+        normalized_mvs = (mvs - mean_m) / stdv
+        # batch_shape x s_M
+
+        cdf_mvs = normal.cdf(normalized_mvs).clamp_min(CLAMP_LB)
+        pdf_mvs = torch.exp(normal.log_prob(normalized_mvs))
+        ratio = pdf_mvs / cdf_mvs
+        check_no_nans(ratio)
+
+        # prepare squared correlation between current and target fidelity
+        rhos_squared = torch.pow(covar_mM.squeeze(-1), 2) / (variance_m * variance_M)
+        # batch_shape x 1
+        check_no_nans(rhos_squared)
+
+        # calculate quality contribution to the GIBBON acqusition function
+        inner_log = 1 - rhos_squared * ratio * (normalized_mvs + ratio)
+        acq = -0.5 * inner_log.clamp_min(CLAMP_LB).log()
+        # average over Gumbel samples
+        acq = acq.mean(dim=1)
+
+        if self.X_pending is not None:
+            # for q>1 GIBBON requires repulsion terms
+            # r_i = log |C_i| for the predictive
+            # correlation matricies C_i between each candidate point in X and
+            # the m current batch elements in X_pending.
+
+            # Each predictive covariance matrix can be expressed as
+            # V_i = [[v_i, A_i], [A_i,B]] for a shared m x m tensor B.
+            # So we can efficientely calculate |V_i| using the formula for
+            # determinant of block matricies
```

Comment:
So at the end of the day this is pretty much the same as what @wjmaddox  is doing on https://github.com/cornellius-gp/gpytorch/pull/1499. It would be nice if we could leverage that PR for computing things here. 

```
base_mat = self.model.posterior(
    self.X_pending, observation_noise=True
)
cross_mat = full_covariance.lazy_covariance_matrix[..., :X.shape[-2], X.shape[-2]:]
new_mat = variance_m. # make this a tensor...
updated_mat = base_mat.cat_rows(cross_mat, new_mat, generate_roots=False)
```

@wjmaddox will your PR give a cheap way to compute the logdet here as well? We'll need to make sure this works in batch mode so we can add the individual `v_i`s as one batch tensor.

Reply:
Though this is also something we can switch over to later and for now do things manually here. 

At any rate, rather than 
```
B_inv = torch.inverse(B)
covariance_determinant = v - torch.matmul(
    torch.matmul(A, B_inv), A.transpose(1, 2)
).squeeze(-1)
```

you should use 
```
covariance_determinant = v - B.inv_quad(A.transpose(1, 2))
```

Reply:
It should be already tested in batch mode (providing gpytorch unit tests for fantasization do that) and you should be able to get a fast log det from it via something like `2 * updated_mat.root_decomposition().root.diag().log().sum(-1)` as long as you regenerate the roots (provided the base_mat matrix is small enough to have a cholesky root decomposition).

Reply:
That is, the log determinant is computed via reading off the diagonals of the updated cholesky (root) factorization.

Reply:
Which, I think just calling `updated_mat.logdet()` should do, but we might want to doublecheck.